Feature/viper config

AppleGamer22 opened the pull request.

AppleGamer22 committed:
install viper, initial config struct

AppleGamer22 committed:
yaml tags for config struct

AppleGamer22 committed:
yaml -> ini

AppleGamer22 committed:
bind flags to viper config

AppleGamer22 committed:
flags variable are now stored in the global cfg struct

AppleGamer22 committed:
cmd args binding to viper

amir1001 reviewed (commented):
Please make sure to test this parameters feeding logic extensively:

1. First priority for reading parameters and arguments - from CLI.
2. Second priority: configuration file.
3. Otherwise: report an error during init.

AppleGamer22 committed:
cfg.ReferenceTimeString without cfg.ReferenceTime

AppleGamer22 committed:
readConfigFile error wrap

AppleGamer22 committed:
arguments overwrite config files

AppleGamer22 committed:
fix: viper key names

viper key names are no longer file type specific, conform to Go's naming scheme
